src/server/transaction.cc

```diff
@@ -577,14 +579,15 @@ OpStatus Transaction::ScheduleSingleHop(RunnableType cb) {
     // The problematic flow is as follows: ScheduleUniqueShard schedules into TxQueue and then
     // call PollExecute that runs the callback which calls DecreaseRunCnt.
     // As a result WaitForShardCallbacks below is unblocked.
-    auto schedule_cb = [this] {
-      bool run_eager = ScheduleUniqueShard(EngineShard::tlocal());
-      if (run_eager) {
+    auto schedule_cb = [&] {
```

Comment:
I literally changed the capture some time ago to a pure this, because the comment above states that we cannot access the stack. Your tests fail on a stack smash